src/interface/swarm_container.cpp

```diff
@@ -36,36 +36,6 @@ void SwarmContainer::Initialize(const std::shared_ptr<StateDescriptor> resolved_
   }
 }
 
-void SwarmContainer::InitializeBoundaries(const std::shared_ptr<MeshBlock> pmb) {
-  if (swarmVector_.empty()) {
-    // No Swarms in this container, so no need to initialize boundaries
-    // This allows default reflecting boundary conditions to be used when no
-    // swarms are present in a parthenon calculation.
-    // NOTE SwarmContainer::Initialize must have already been called.
-    return;
-  }
-
-  std::stringstream msg;
-  auto &bcs = pmb->pmy_mesh->mesh_bcs;
-  // Check that, if we are using user BCs, they are actually enrolled, and unsupported BCs
-  // are not being used
-  for (int iFace = 0; iFace < 6; iFace++) {
-    if (bcs[iFace] == BoundaryFlag::user) {
-      if (pmb->pmy_mesh->forest.GetTreePtr(pmb->loc.tree())->SwarmBndryFnctn[iFace] ==
-          nullptr) {
-        msg << (iFace % 2 == 0 ? "i" : "o") << "x" << iFace / 2 + 1
-            << " user boundary requested but provided function is null!";
-        PARTHENON_FAIL(msg);
```

Comment:
@lroberts36 With the removal of boundary functions from the  mesh, this check no longer passes on a forest with two trees in the X1 direction and user swarm boundary functions enrolled for `ix1` and `ox1` -- basically each tree has only one of those two BCs but this check tries to find both BCs on each tree. I think it is fine to just remove this check because each tree will only call the boundaries appropriate for that tree (for example a tree with only block boundaries on its `ox1` side should not call user `ox1` boundary conditions on the swarms). I wanted to check with you though that this is reasonable to do.

Reply:
Yes, I think this is reasonable. As you say, trees that share a boundary with another tree will not have physical boundary conditions defined on those boundaries.